shell/components/AsyncButton.vue

```diff
@@ -246,16 +249,21 @@ export default defineComponent({
       }
 
       const cb: AsyncButtonCallback = (success) => {
-        this.done(success);
+        this.done(success, ev);
       };
 
       this.$emit('click', cb);
     },
 
-    done(success: boolean | 'cancelled') {
+    done(success: boolean | 'cancelled', ev?: any) {
       if (success === 'cancelled') {
         this.phase = ASYNC_BUTTON_STATES.ACTION;
       } else {
+        // ev.detail = 0 means that the trigger was a key press
+        if (!success && this.returnFocusOnError && ev && ev.detail === 0) {
+          // set the focus back on the async button. We lose it when saving a "form" HTML element
+          setTimeout(() => this.focus(), 0);
```

Comment:
I'm not 100% sure how I feel about this approach. A user can shift focus to different inputs while waiting for an async button to complete. We will steal focus back to the button in the event of an error. I can see this being frustrating in the following scenario:

1. I improperly fill out a form 
2. Click on the async button to submit
3. While the async button is waiting, I realize that I made an error and focus an input to make a correct
4. The async button enters an error state and steals focus away from me while I'm trying to make a correction